src/lib.rs

```diff
@@ -454,12 +502,53 @@ impl<'object, Endian> DebugInfo<'object, Endian>
                 // matches us better. We need to keep going until we find the best one.
             }
 
-            return Ok(Some((path, line, func.map(|u| u.0.name.to_string_lossy()))));
+            let func = func.map(|u| {
+                if unit.language.is_some() {
+                    debug_assert!(self.opts.with_demangling,
+                                  "We shouldn't even bother finding the DW_AT_language if we \
+                                   aren't demangling");
+                }
+                match unit.language {
+                    Some(gimli::DW_LANG_C_plus_plus) |
+                    Some(gimli::DW_LANG_C_plus_plus_03) |
+                    Some(gimli::DW_LANG_C_plus_plus_11) => demangle_cpp_symbol(u.0.name),
+                    Some(gimli::DW_LANG_Rust) => demangle_rust_symbol(u.0.name),
+                    _ => u.0.name.to_string_lossy(),
```

Comment:
Yeah I'm not sure what the best way to do that would be. Maybe if we added `env_logger` or something, but I don't think that is necessarily a good idea...